z2/src/validators.rs

```diff
@@ -164,6 +170,17 @@ fn hex_string_to_u8_20(hex_str: &str) -> Result<[u8; 20], &'static str> {
     Ok(array)
 }
 
+pub fn pop_prove(secret_key: &str) -> Result<()> {
```

Comment:
Not sure where is best to compute the POP signature.
1. Add a new `z2 pop` command that computes the pop-signature given a secret-key; or
2. Alter the `deposit` command to take a secret-key instead, and compute both public-key and pop-signatures from it; or
3. Provide an external tool/script that computes the pop-signature given a secret-key; or
4. All the above.

Reply:
@chetan-zilliqa Any opinions? I can see arguments for both 1 or 2.

Reply:
Okay, so with option 1, the PoP signature is created separately, and the validator operator must pass it as a parameter to the z2 deposit command. In option 2, both the public-key and pop-signature would be generated internally by the tool, eliminating the need to pass the --public-key parameter to z2 deposit. This simplifies the process for the node operator by removing an additional step. Option 2 seems more efficient to me. However, it appears that z2 deposit currently accepts the secret key as an argument, so it might not require any changes. Apologies if I've misunderstood or caused any confusion

Reply:
@chetan-zilliqa for option 2, the `z2 deposit` command currently only accepts the public-key. So, it would require a change to take the private-key as an input.

As for options 3 & 1, it might also be feasible to modify the `convert-key` tool to dump out the public-key and pop-signature instead.

Either way, it's a small change. So, just let me know what's the preference.

Reply:
@chetan-zilliqa I understand what you mean now with #1234 but that is still in a draft stage. I guess that if you intend to ask users to supply an actual `--private-key` to the `z2 deposit` command, then I guess you can adapt the code from `convert-key.rs` to generate the `pop-signature` as well.

Reply:
It's ok @shawn-zil . Yes. I am open to it.